File: backend/app/schemas/blog.py
from pydantic import BaseModel
from typing import Optional, List
import uuid
from datetime import datetime

class GenerateRequest(BaseModel):
    query: str
    source_url: Optional[str] = None

class BlogOut(BaseModel):
    id: uuid.UUID
    brand_id: uuid.UUID
    title: str
    content: str
    meta_description: Optional[str]
    image_url: Optional[str]
    source_url: Optional[str]
    keyword_targets: Optional[List[str]]
    word_count: Optional[int]
    created_at: datetime

    class Config:
        from_attributes = True

class BlogListItem(BaseModel):
    id: uuid.UUID
    title: str
    meta_description: Optional[str]
    image_url: Optional[str]
    word_count: Optional[int]
    created_at: datetime

    class Config:
        from_attributes = True

class BlogListResponse(BaseModel):
    blogs: List[BlogListItem]
    total: int

File: frontend/src/types/blog.ts
export interface Blog {
  id: string;
  brand_id: string;
  title: string;
  content: string;
  meta_description: string;
  image_url: string | null;
  source_url: string | null;
  keyword_targets: string[];
  word_count: number;
  created_at: string;
}

export interface BlogListItem {
  id: string;
  title: string;
  meta_description: string;
  image_url: string | null;
  word_count: number;
  created_at: string;
}

export interface BlogListResponse {
  blogs: BlogListItem[];
  total: number;
}

export interface GenerateRequest {
  query: string;
  source_url?: string;
}
